### PR TITLE
Update project to Swift 4

### PR DESCRIPTION
**Pull request by onmyway133**
- Update project to swift 4
- Point to swift 4 branch of `Cache` for now, as new version is not released yet.

### CODE REVIEW

File: Sources/Shared/Configuration.swift [hunk 1]
```diff
@@ -3,12 +3,11 @@ import Cache
...
 public struct Configuration {
   public static var bytesLoaded: Int = 0
-  public static var imageCache: SpecializedCache<Image> = {
-    let config = Config(
-        expiry: Expiry.date(NSDate().addingTimeInterval(60 * 60 * 24 * 3) as Date),
-        memoryCountLimit: 10,
-        maxDiskSize: 0)
 
-    return SpecializedCache<Image>(name: "Imaginary", config: config)
+  public static var imageStorage: Storage? = {
```
Comment (onmyway133) [inline Sources/Shared/Configuration.swift:7]: How do you prefer to handle error in this case @hyperoslo/ios ?

File: Sources/Shared/ImageManager.swift [hunk 2]
```diff
@@ -26,15 +27,16 @@ public class ImageManager {
...
-    Configuration.imageCache.async.object(forKey: url.absoluteString) { [weak self] (object: Image?) in
+    Configuration.imageStorage?.async.object(ofType: ImageWrapper.self,
+                                            forKey: url.absoluteString) { [weak self] result in
       guard let `self` = self else {
         return
       }
 
       // Return image from cache if it exists.
-      if let image = object {
+      if case .value(let wrapper) = result {
```
Comment (vadymmarkov) [inline Sources/Shared/ImageManager.swift:37]: I think we it's good to call completion in the case of error as well.

File: Sources/Shared/ImageManager.swift [hunk 3]
```diff
@@ -69,7 +71,10 @@ public class ImageManager {
       case let .success(image, bytes):
         Configuration.track?(url, nil, bytes)
         if useCache {
-          Configuration.imageCache.async.addObject(image, forKey: url.absoluteString)
+          let wrapper = ImageWrapper(image: image)
+          Configuration.imageStorage?.async.setObject(wrapper, forKey: url.absoluteString, expiry: nil) { _ in
```
Comment (vadymmarkov) [inline Sources/Shared/ImageManager.swift:75]: Shouldn't completion be optional?

**Review by vadymmarkov — APPROVED** [review]

File: Sources/Shared/ImageManager.swift (hunk 3 above)
Comment (onmyway133) [reply Sources/Shared/ImageManager.swift:75]: @vadymmarkov: no, this callback can't be both `@escaping` and optional

File: Sources/Shared/ImageManager.swift (hunk 2 above)
Comment (onmyway133) [reply Sources/Shared/ImageManager.swift:37]: @vadymmarkov: The signature is `public typealias Completion = (Image?) -> Void` so I think we should fix that in the next PR

File: Sources/Shared/ImageManager.swift (hunk 3 above)
Comment (vadymmarkov) [reply Sources/Shared/ImageManager.swift:75]: But from public API perspective it's not very nice to always have completion.

**Pull request merged**